client/src/components/screens/PlannerPage/EventCard.js

```diff
@@ -32,10 +32,26 @@ const useStyles = makeStyles(() => ({
 export default function EventCard({ event, setDeleteFlag }) {
     const classes = useStyles();
     const [isEditable, toggleEdit] = useState(false);
+    const [isSave, toggleSave] = useState(false);
     const [originalEvent, changeOriginalEvent] = useState(event);
     const [selectedEvent, changeSelectedEvent] = useState(event);
     const history = useHistory();
 
+    const initCachedRoles = (event) => {
```

Comment:
Does clone deep not work here?

Reply:
Yeah, this part is actually redundant. I changed it in the recent commit. Let me know if there are any other parts worth refactoring.